Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -183,10 +184,43 @@ namespace ROS2
         // Build up a list of all entities created as a part of processing the file.
         AZStd::vector<AZ::EntityId> createdEntities;
         AZStd::unordered_map<const sdf::Link*, AzToolsFramework::Prefab::PrefabEntityResult> createdLinks;
+        AZStd::unordered_map<const sdf::Model*, AzToolsFramework::Prefab::PrefabEntityResult> createdModels;
         AZStd::unordered_map<AZStd::string, const sdf::Link*> links;
-        for ([[maybe_unused]] const auto& [fullLinkName, linkPtr, _] : linksMapper.m_links)
+        for ([[maybe_unused]] const auto& [fullLinkName, linkPtr, attachedModel] : linksMapper.m_links)
         {
-            createdLinks[linkPtr] = AddEntitiesForLink(linkPtr, AZ::EntityId{}, createdEntities);
+            // Create entities for the model containing the link
+            AZ::EntityId modelEntityId;
+            if (attachedModel != nullptr)
+            {
+                if (auto modelIt = createdModels.find(attachedModel); modelIt != createdModels.end())
+                {
+                    if (modelIt->second.IsSuccess())
+                    {
+                        modelEntityId = modelIt->second.GetValue();
+                    }
+                }
+                else
+                {
+                    if (AzToolsFramework::Prefab::PrefabEntityResult createModelEntityResult = CreateEntityForModel(*attachedModel);
+                        createModelEntityResult)
+                    {
+                        modelEntityId = createModelEntityResult.GetValue();
+                        // Add the model entity to the created entity list
+                        // so that it gets added to the prefab
+                        createdEntities.emplace_back(modelEntityId);
+
+                        std::string modelName = attachedModel->Name();
+                        AZStd::string azModelName(modelName.c_str(), modelName.size());
+                        AZStd::lock_guard<AZStd::mutex> lck(m_statusLock);
+                        m_status.emplace(azModelName, AZStd::string::format("created as: %s", modelEntityId.ToString().c_str()));
```

Comment:
Huh, I'm surprised we never added a ToFixedString() for EntityId.

Reply:
I guess we never had a pass done over it.
Since it is a 64-bit value, it can fit within 20-digits, so an `AZStd::fixed_string<32>` would be the best data type for it, if we do add the overload.